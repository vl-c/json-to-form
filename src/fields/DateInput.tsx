import * as React from 'react';

export interface DateInputProps {
  id: string;
  label?: string;
  value?: Date;
}

const DateInput = (props: DateInputProps): JSX.Element => {
  const { id, label, value = new Date() } = props;
  // better use datepickers and moment/another date libs
  // without timezone, just for example
  const dateValue = value.toISOString().split('T')[0];

  return (
    <div className="input-block">
      {label && (
        <label className="input-label" htmlFor={id}>
          {label}
        </label>
      )}
      <input
        className="input-field date-field"
        id={id}
        name={id}
        type="date"
        defaultValue={dateValue}
      />
    </div>
  );
};

export default DateInput;
