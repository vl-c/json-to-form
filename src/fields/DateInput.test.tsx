import * as React from 'react';

import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import DateInput from './DateInput';

function renderDateInput() {
  return shallow(
    <DateInput id={'test'} label={'DateInput'} value={new Date()} />
  );
}

it('DateInput shallow render', () => {
  const res = renderDateInput();
  expect(shallowToJson(res)).toMatchSnapshot();
});
