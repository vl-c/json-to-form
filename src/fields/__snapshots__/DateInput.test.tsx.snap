// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateInput shallow render 1`] = `
<div
  className="input-block"
>
  <label
    className="input-label"
    htmlFor="test"
  >
    DateInput
  </label>
  <input
    className="input-field date-field"
    defaultValue="2021-04-13"
    id="test"
    name="test"
    type="date"
  />
</div>
`;
